### PR TITLE
merge in branch before develop / master

### PR DESCRIPTION
**Pull request by ischer**
i'm proposing we consider switching to a different flow for git finish.

right now, we just checkout master, pull, and merge our working branch in. this could be somewhat problematic if there are conflicts. it would force us to resolve conflicts and finish the merge manually in master (or develop). if the conflicts are messy, this could potentially lead to buggy code getting committed to master.

i'm suggesting that git finish first checkout and pull on both develop and master, then perform the merge in the current working branch. this way, if someone runs into merge conflicts, they can resolve them and test them in their own branch, rather than develop or master. after merging in the working branch, it would then checkout develop / master, pull again (just to be sure), and merge the working branch. unless something crazy happens, this should prevent us from having to resolve conflicts in master or develop.

let me know what you think.

### CODE REVIEW

**Comment by jasonrhodes** [comment]
I somehow missed the notification when this got created, but I'm going to look at it today. Thanks @ischer

**Comment by jasonrhodes** [comment]
OK so I think the way forward here is for you or someone to show what the commit history looks like the normal way vs using this to merge in first. I like the idea but don't want it to get messy in the history ... just want to confirm that it won't.

File: commands/git-finish [hunk 1]
```diff
@@ -29,19 +29,39 @@ if [[ $BRANCH = 'master' || $BRANCH = 'develop' ]]; then
...
   git checkout master
   git pull origin master
-  git merge --no-ff $BRANCH
-  [[ $1 = 'push' || $2 = 'push' ]] && git push origin master
+fi
+
+git checkout $BRANCH
+
+if [[ $DEVELOP_EXISTS -eq 0 ]]; then
+  git merge --no-ff develop
```
Comment (jasonrhodes) [inline commands/git-finish:45]: If we do this maybe these (2) pre-merges should be done without `--no-ff`? Need to see the different histories to know for sure.

**Comment by jasonrhodes** [comment]
Also I can help with the merge conflicts, I just slightly changed how we check for DEVELOP existing because the way I was doing it before had a small edge case bug.

**Comment by ischer** [comment]
didn't see notifications for your comments on this - i'll report back with the history of this vs without this soon. if it gets messy, maybe we could squash some bits.

**Pull request closed without merging**